securesystemslib/signer/_gpg_signer.py

```diff
@@ -35,41 +240,78 @@ def from_priv_key_uri(
         public_key: Key,
         secrets_handler: Optional[SecretsHandler] = None,
     ) -> "GPGSigner":
-        raise NotImplementedError("Incompatible with private key URIs")
+        if not isinstance(public_key, GPGKey):
+            raise ValueError(f"expected GPGKey for {priv_key_uri}")
+
+        uri = parse.urlparse(priv_key_uri)
+
+        if uri.scheme != cls.SCHEME:
+            raise ValueError(f"GPGSigner does not support {priv_key_uri}")
+
+        if secrets_handler is not None:
+            raise ValueError("GPGSigner does not support a secrets handler")
+
```

Comment:
I think this is not useful -- code that uses from_priv_key_uri() is aiming to work with as many signers as it can, using the shared API. This seems to force me to write two code paths for no gain.

Reply:
@jku, do you think we should just ignore it? Maybe log a warning/info? With current implementation of `securesystemslib.gpg` we cannot pass any secrets to the gpg command.

Reply:
No need for warning as far as I am concerned: the way I imagine this being used (and the way I am using it) is that the signing code doesn't care which signers are used or what they do:
* online signing app likely does not have a secrets handler
* an offline signing tool does provide a secrets handler
* even the offline signing tool _does not care whether the secrets handler is used or not_: a gpg signing might not use it but a yubikey signing will